src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -95,18 +125,389 @@ public PopulationSimulationPKAnalyses CalculateFor(PopulationSimulation populati
          if (!populationSimulation.HasResults)
             return new NullPopulationSimulationPKAnalyses();
 
-         var bodyWeightParameter = populationSimulation.BodyWeight;
-         var bodyWeightParameterPath = bodyWeightParameterPathFrom(bodyWeightParameter);
-         var allBodyWeights = populationSimulation.AllValuesFor(bodyWeightParameterPath);
-
          try
          {
-            return base.CalculateFor(populationSimulation, populationSimulation.Results, (individualId) => { updateBodyWeightFromCurrentIndividual(bodyWeightParameter, allBodyWeights, individualId); });
+            var globalPKAnalysesForIndividuals = new Cache<int, GlobalPKAnalysis>();
+            var analyses = base.CalculateFor(populationSimulation, populationSimulation.Results,
+               individualId =>
+               {
+                  _populationSimulationBodyWeightUpdater.UpdateBodyWeightForIndividual(populationSimulation, individualId);
+               },
+               (individualId, options, compoundName) =>
+               {
+                  var globalPKAnalysis = globalPkAnalysisForIndividual(globalPKAnalysesForIndividuals, individualId);
+                  globalPKAnalysis.Add(calculateGlobalPKParameters(populationSimulation, individualId, compoundName, options, globalPKAnalysis));
+               });
+
+            mapQuantityPKParametersFromIndividualGlobalPKAnalyses(globalPKAnalysesForIndividuals.ToList()).Each(x => analyses.AddPKAnalysis(x));
```

Comment:
why the ToList. It's already an enumerable
globalPKAnalysesForIndividuals.Each(analyses.AddPKAnalysis)